### PR TITLE
EC2 Instance Connect Endpoint: HTTP endpoint to create Nodes

### PR DESCRIPTION
**Pull request by marcoandredinis**
When the user receives a list of EC2 instances, they must be able to pick one and add it to the cluster.

To do so, we must provide an API that accepts the node information and adds it to the backend.

Only Nodes of type SubKindOpenSSHEC2InstanceConnectEndpointNode are allowed. Other nodes have their own join method which is not part of the HTTP API.

Demo:
After calling the new endpoint, the node is visible in the UI:
image

### CODE REVIEW

File: lib/web/servers.go [hunk 1]
```diff
@@ -335,3 +336,84 @@ func getDatabaseUsersAndNames(accessChecker services.AccessChecker) (dbNames []s
...
+	if err := req.checkAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clt, err := sctx.GetUserClient(ctx, site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	labels := make(map[string]string)
```
Comment (nklaassen) [inline lib/web/servers.go:393]:
```suggestion
	labels := make(map[string]string, len(req.Labels))
```

File: lib/web/servers_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+	ctx := context.Background()
+	username := "someuser"
+	roleUpsertNode, err := types.NewRole(services.RoleNameForUser(username), types.RoleSpecV6{
+		Allow: types.RoleConditions{
+			Logins: []string{"osuser"},
+			Rules: []types.Rule{
+				types.NewRule(types.KindNode,
+					[]string{types.VerbCreate, types.VerbUpdate}),
+			},
```
Comment (nklaassen) [inline lib/web/servers_test.go:43]: could you add a test that an unauthorized user cannot create the node? Just to make sure we always use the right auth client

**Review by nklaassen — APPROVED** [review]

File: lib/web/servers.go [hunk 3]
```diff
@@ -335,3 +336,84 @@ func getDatabaseUsersAndNames(accessChecker services.AccessChecker) (dbNames []s
...
+	if r.Name == "" {
+		return trace.BadParameter("missing node name")
+	}
+
+	// Nodes provided by the Teleport Agent are not meant to be created by the user.
+	// They connect to the cluster and heartbeat their information.
+	//
+	// Agentless Nodes with Teleport CA call the Teleport Proxy and upsert themselves,
+	// so they are also not meant to be added from web api.
+	if r.SubKind != types.SubKindOpenSSHEC2InstanceConnectEndpointNode {
```
Comment (r0mant) [inline lib/web/servers.go:360]: `SubKindOpenSSHEC2InstanceConnectEndpointNode` lol this is a mouthful, how about renaming this to `SubKindEIC` (I know it's not a part of this PR but still)?

**Review by r0mant — APPROVED** [review]

File: lib/web/servers_test.go (hunk 2 above)
Comment (marcoandredinis) [reply lib/web/servers_test.go]: Done :pray:

File: lib/web/servers.go (hunk 3 above)
Comment (marcoandredinis) [reply lib/web/servers.go:360]: What about `SubKindOpenSSHEIC`?
I think keeping the OpenSSH part is useful to give context, the rest is just a product/marketing name

Comment (r0mant) [reply lib/web/servers.go]: `SubKindOpenSSHEICE` sounds good to me.

Comment (marcoandredinis) [reply lib/web/servers.go]: I'll go with `SubKindOpenSSHEICE`
Has an extra `E`, but matches the actual feature name: EC2 Instance Connect Endpoint

Let me know if you prefer the other option and I'll change it in the other PR

**Pull request merged**